src/llm/apis/openai_completions.cpp

```diff
@@ -224,30 +237,58 @@ absl::Status OpenAIChatCompletionsHandler::parseMessages() {
+                                try {
+                                    ov::Tensor tensor = loadImageStbiFromMemory(decoded);
+                                    request.imageHistory.push_back({i, tensor});
+                                } catch (std::runtime_error& e) {
+                                    std::stringstream ss;
+                                    ss << "Image parsing failed: " << e.what();
+                                    SPDLOG_LOGGER_ERROR(llm_calculator_logger, ss.str());
+                                    return absl::InvalidArgumentError(ss.str());
+                                }
                             } else if (std::regex_match(url.c_str(), std::regex("^(https?:\\/\\/)?([\\da-z\\.-]+)\\.([a-z\\.]{2,6})([\\/\\w \\.-]*)*\\/?$"))) {
+                                SPDLOG_LOGGER_TRACE(llm_calculator_logger, "Loading image using curl");
                                 curl_global_init(CURL_GLOBAL_ALL);
                                 int64_t sizeLimit = 20000000;  // restrict single image size to 20MB
                                 auto status = downloadImage(url.c_str(), decoded, sizeLimit);
                                 if (status != absl::OkStatus()) {
                                     return status;
                                 }
                                 curl_global_cleanup();
+                                try {
+                                    ov::Tensor tensor = loadImageStbiFromMemory(decoded);
+                                    request.imageHistory.push_back({i, tensor});
+                                } catch (std::runtime_error& e) {
+                                    std::stringstream ss;
+                                    ss << "Image parsing failed: " << e.what();
+                                    SPDLOG_LOGGER_ERROR(llm_calculator_logger, ss.str());
+                                    return absl::InvalidArgumentError("Image parsing failed");
+                                }
+
                             } else {
-                                return absl::InvalidArgumentError("Url should contain base64 encoded string followed by \"base64,\" prefix or valid URL");
-                            }
-                            try {
-                                ov::Tensor tensor = load_image_stbi(decoded);
-                                request.imageHistory.push_back({i, tensor});
-                            } catch (std::runtime_error& e) {
-                                std::stringstream ss;
-                                ss << "Image parsing failed: " << e.what();
-                                SPDLOG_LOGGER_ERROR(llm_calculator_logger, ss.str());
-                                return absl::InvalidArgumentError("Image parsing failed");
+                                SPDLOG_LOGGER_TRACE(llm_calculator_logger, "Loading image from local filesystem");
+                                if (allowedLocalMediaPath.has_value()) {
```

Comment:
If allow local media path does not exist we continue and read url, so we allow everything then, is that intended?